src/RemoteConfigContext.tsx

```diff
@@ -83,49 +86,35 @@ export const RemoteConfigContextProvider: React.FC = ({children}) => {
         Bugsnag.notify(e as any);
       }
     }
-  }
+  }, []);
 
   useEffect(() => {
-    async function setupRemoteConfig() {
+    (async function setupRemoteConfig() {
       const configApi = remoteConfig();
-      await configApi.setConfigSettings({
-        minimumFetchIntervalMillis: 21600000, // 6 hours
-      });
+      await configApi.setConfigSettings({minimumFetchIntervalMillis: 21600000}); // 6 hours
       await configApi.setDefaults(defaultRemoteConfig);
-      if (newBuildSincePreviousLaunch) {
-        await refresh();
-      } else {
-        await fetchConfig();
-      }
-    }
-
-    if (!isLoadingAppState) {
-      setupRemoteConfig();
-    }
-  }, [isLoadingAppState, newBuildSincePreviousLaunch]);
+      await fetchConfig();
+    })();
+  }, [fetchConfig]);
 
   useInterval(
     () => {
       fetchConfig();
       incrementRetryInterval();
     },
     retryInterval,
-    undefined,
+    [fetchConfig],
     !fetchError,
   );
 
-  async function refresh() {
+  const refresh = useCallback(async () => {
     const configApi = remoteConfig();
     const {minimumFetchIntervalMillis} = configApi.settings;
-    await configApi.setConfigSettings({
-      minimumFetchIntervalMillis: 0,
-    });
+    await configApi.setConfigSettings({minimumFetchIntervalMillis: 0});
```

Comment:
This will refresh immediately the data, when refresh is called. So the idea is not refresh it, right? Remote config suggest the following patterns https://firebase.google.com/docs/remote-config/loading and one of them is just refresh at the start of the app.

Reply:
The guide suggest using `fetchAndActivate` which we already use at startup. But in addition to that we had the force refresh code which triggered on every build number change. I think the force refresh code is not needed anymore, and we can just rely on `fetchAndActivate` as mentioned by the guide.

The guide also suggest using `addOnConfigUpdateListener`, which I think would be nice. However, I think that would be its own issue to not bloat this small change.